plugin-tester-java/src/test/scala/example/myapp/helloworld/ErrorReportingSpec.scala

```diff
@@ -32,9 +34,14 @@ class ErrorReportingSpec extends AnyWordSpec with Matchers with ScalaFutures wit
 
     val handler = GreeterServiceHandlerFactory.create(new GreeterServiceImpl(mat), sys)
     val binding = {
-      import akka.http.javadsl.{ ConnectHttp, Http }
+      import akka.http.javadsl.Http
+      import akka.http.javadsl.model.{ HttpRequest, HttpResponse }
 
-      Http(sys).bindAndHandleAsync(handler, ConnectHttp.toHost("127.0.0.1", 0), mat).toCompletableFuture.get
+      Http(sys)
+        .newServerAt("127.0.0.1", 0)
+        .bind((req => handler(req)): akka.japi.function.Function[HttpRequest, CompletionStage[HttpResponse]])
```

Comment:
I think we can (and should) make this simpler (#1101), but we can do that in a separate PR